nvflare/app_opt/lightning/api.py

```diff
@@ -34,6 +36,10 @@ def patch(trainer: pl.Trainer):
         callbacks = [callbacks, fl_callback]
     else:
         callbacks = [fl_callback]
+
+    if restore_optimizers:
+        callbacks.append(RestoreOptimizers())
+
```

Comment:
Since we also have "optimizer_params" in the FLModel, I am afraid users might think that this "restore_optimizers" argument will automatically takes in "optimizer_params"?

This "RestoreOptimizers" is actually locally remember something and then load it back.

I think my point is:
It is good for a "lightning" application.
But it needs more enhancement for a "lightning" + FL scenario.

Not sure where is the best place to put "RestoreOptimizers" though

Reply:
Renamed to `RestoreState` and updated docstring to explain we are restoring the local optimizer and lr scheduler states.